skyportal/handlers/api/telescope.py

```diff
@@ -135,23 +135,32 @@ def get(self, telescope_id=None):
         telescopes = []
         for telescope in data:
             temp = telescope.to_dict()
-            if telescope.next_twilight_morning_astronomical() is not None:
-                temp['is_night_astronomical'] = bool(
-                    telescope.next_twilight_morning_astronomical().jd
-                    < telescope.next_twilight_evening_astronomical().jd
-                )
-                temp[
-                    'next_twilight_morning_astronomical'
-                ] = telescope.next_twilight_morning_astronomical().iso
-                temp[
-                    'next_twilight_evening_astronomical'
-                ] = telescope.next_twilight_evening_astronomical().iso
-            else:
-                temp['is_night_astronomical'] = False
-                temp['next_twilight_morning_astronomical'] = False
-                temp['next_twilight_evening_astronomical'] = False
-
+            morning = False
+            evening = False
+            is_night_astronomical = False
+            if (
+                telescope.fixed_location
+                and telescope.lon
+                and telescope.lat
+                and telescope.elevation
+            ):
+                try:
+                    morning = telescope.next_twilight_morning_astronomical()
+                    evening = telescope.next_twilight_evening_astronomical()
+                    if morning is not None and evening is not None:
+                        is_night_astronomical = bool(morning.jd < evening.jd)
+                        morning = morning.iso
+                        evening = evening.iso
+                except Exception:
+                    morning = False
```

Comment:
Since you set the default above, you do not need to reset morning and such right?

Reply:
If the exception happens for the evening, i want the morning to also be false, so i reset it all.